dask/dataframe/core.py

```diff
@@ -3684,27 +3667,34 @@ def map_partitions(func, *args, **kwargs):
         else:
             args2.append(arg)
 
-    kwargs_task, deps = unpack_collections(kwargs2)
-    if deps:
-        dependencies.extend(deps)
-    else:
-        kwargs_task = kwargs2
-
+    # Normalize keyword arguments
+    kwargs3 = {}
+    from ..array.core import normalize_arg
+    for k, v in kwargs.items():
```

Comment:
Should this be kwargs2? (I don't really know what kwargs2 is, but that's what was deleted)

Reply:
Nice catch.  I've cleaned this up a bit.